### PR TITLE
PHOENIX-6203 : Add new CQS method to return Table instance only if table exists

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableIT.java [hunk 1]
```diff
@@ -1440,5 +1441,69 @@ public void testAddNonPKColumnWhenlastPKIsVARBINARYOrARRAY() throws Exception {
...
+            Table t2 = cqs.getTable(Bytes.toBytes(tableName2));
+            assertEquals(t2.getName().getNameAsString(), tableName2);
+
+            // this is correct check for existence of table
+            try {
+                cqs.getTableIfExists(Bytes.toBytes(tableName1));
+                fail("Should have thrown TableNotFoundException");
+            } catch (TableNotFoundException e) {
+                // exact match is possible because schema name is empty
+                assertEquals(e.getTableName(), tableName1);
```
Comment (yanxinyi) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableIT.java:1496]: nit: the first argument should be the expected value.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableIT.java [hunk 2]
```diff
@@ -1440,5 +1441,69 @@ public void testAddNonPKColumnWhenlastPKIsVARBINARYOrARRAY() throws Exception {
...
+            }
+            try {
+                cqs.getTableIfExists(Bytes.toBytes(tableName2));
+                fail("Should have thrown TableNotFoundException");
+            } catch (TableNotFoundException e) {
+                // just asserting table name, not schema name
+                assertTrue(tableName2.contains(e.getTableName()));
```
Comment (yanxinyi) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableIT.java:1503]: nit: can we use the same way as above, which is assertEquals?

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  8s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 35s |  master passed  |
| +1 :green_heart: |  compile  |   1m  1s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 48s |  master passed  |
| +0 :ok: |  spotbugs  |   3m  6s |  phoenix-core in master has 962 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m  4s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 18s |  phoenix-core: The patch generated 6 new + 3059 unchanged - 2 fixed = 3065 total (was 3061)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 21s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 157m 11s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 23s |  The patch does not generate ASF License warnings.  |
|  |   | 197m 34s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.TableSnapshotReadsMapReduceIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/957 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux e15fcabac435 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 859b16c |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/1/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/1/testReport/ |
| Max. process+thread count | 6177 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  8s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 18s |  master passed  |
| +1 :green_heart: |  compile  |   1m  2s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 45s |  master passed  |
| +0 :ok: |  spotbugs  |   3m 11s |  phoenix-core in master has 962 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 20s |  phoenix-core: The patch generated 6 new + 3059 unchanged - 2 fixed = 3065 total (was 3061)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 18s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 159m 44s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 23s |  The patch does not generate ASF License warnings.  |
|  |   | 199m 50s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/957 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 62041d9e2d6e 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 859b16c |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/3/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/3/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/3/testReport/ |
| Max. process+thread count | 6199 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-957/3/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by yanxinyi — APPROVED** [review]

**Comment by virajjasani** [comment]
Thanks for the review @yanxinyi

**Pull request closed without merging**